patientsearch/src/js/components/PatientListTable.js

```diff
@@ -264,182 +294,111 @@ export default function PatientListTable() {
   const getISS = function () {
     return appSettings["SOF_HOST_FHIR_URL"];
   };
-  const getLaunchURL = function (patientId) {
+  const getLaunchURL = function (patientId, launchParams) {
+    launchParams = launchParams || {};
+    const LAUNCH_URL = "SOF_CLIENT_LAUNCH_URL";
     if (!patientId) {
       console.log("Missing information: patient Id");
       return "";
     }
-    let baseURL = getLaunchBaseURL();
+    let baseURL = launchParams[LAUNCH_URL] ? launchParams[LAUNCH_URL] : getLaunchBaseURL();
     let iss = getISS();
     if (!baseURL || !iss) {
       console.log("Missing ISS launch base URL");
       return "";
     }
-    let launchParam = btoa(JSON.stringify({ b: patientId }));
-    return `${baseURL}?launch=${launchParam}&iss=${iss}`;
+    return `${baseURL}?launch=${btoa(JSON.stringify({ b: patientId }))}&iss=${iss}`;
   };
-  const noCacheParam = { cache: "no-cache" };
-
-  const existsIndata = function (rowData) {
-    if (!data) return false;
-    if (!rowData) return false;
-    return (
-      data.filter((item) => {
-        return parseInt(item.id) === parseInt(rowData.id);
-      }).length > 0
-    );
+  const hasMultipleSoFClients = () => {
+    return launchInfos && launchInfos.length > 1;
   };
-  const addDataRow = function (rowData) {
-    if (!rowData || !rowData.id) return false;
-    let newData = formatData(rowData);
-    if (newData && !existsIndata(newData[0])) {
-      setData([newData[0], ...data]);
+  const launchAPP = (rowData, launchParams) => {
+
+    setCurrentRow(rowData);
+
+    console.log("has multiple launch apps ? ", hasMultipleSoFClients());
+    console.log("launch params ", launchParams);
+
+    //handle multiple SoF clients that can be launched
+    //open a dialog here?
```

Comment:
Yes, we can expect multiple SoF clients, and each may have it's own `REQUIRED_ROLES` and should only be limited to user having such a role in their JWT if `REQUIRED_ROLES` is non empty for each respective SoF client.